drivers/QSPI.h

```diff
@@ -20,20 +20,48 @@
 
 #if defined (DEVICE_QSPI) || defined(DOXYGEN_ONLY)
 
-#include "platform/PlatformMutex.h"
 #include "hal/qspi_api.h"
+#include "platform/PlatformMutex.h"
 #include "platform/SingletonPtr.h"
 #include "platform/NonCopyable.h"
 
-#define QSPI_DEFAULT_INST_WIDTH       QSPI_CFG_BUS_SINGLE //Single bit mode for Instruction as most devices use 1-4-4 mode
-#define QSPI_DEFAULT_ADDRESS_WIDTH    QSPI_CFG_BUS_QUAD //QuadSPI mode
-#define QSPI_DEFAULT_ADDRESS_SIZE     QSPI_CFG_ADDR_SIZE_32
-#define QSPI_DEFAULT_ALT_WIDTH        QSPI_CFG_BUS_QUAD //QuadSPI mode 
-#define QSPI_DEFAULT_ALT_SIZE         QSPI_CFG_ALT_SIZE_NONE
-#define QSPI_DEFAULT_DATA_WIDTH       QSPI_CFG_BUS_QUAD //QuadSPI mode
-#define QSPI_DEFAULT_DUMMY_CYCLES     0
-#define _1_MHZ_                       1000000 
-#define QSPI_DEFAULT_HZ               _1_MHZ_
+#define ONE_MHZ     1000000
+
+/** QSPI Bus width Enum
+ */
+typedef enum qspi_config_bus_width {
```

Comment:
I would limit the scope of these definitions, to add them inside QSPI class scope

Reply:
I wonder if this can make it inconvenient for programmers to use scope resolution operator on return checks as we are using qspi_return_status_t as our return code for all functions. So for now, I'm going to this inside mbed namespace but outside of class definition.